ts/screens/wallet/payment/ScanQrCodeScreen.tsx

```diff
@@ -196,6 +217,11 @@ class ScanQrCodeScreen extends React.Component<Props, State> {
     switch (barcode.format) {
       case "QRCODE":
         this.onQrCodeData(barcode.value);
+        break;
+
+      case "DATA_MATRIX":
+        this.onDataMatrixData(barcode.value);
+        break;
```

Comment:
With the new current extension we are now able to scan a data matrix using the camera.
Shouldn't the user be able to upload a data matrix also from the gallary? 
I'm asking because I don't know if is required / planned or is out of scope of this story, I've tried to add a data matrix from the gallery and it's not recognized.

Reply:
It's a really valid point, we didn't take it into consideration but I think it's out of this ticket's scope. I'm going to investigate a possible solution for this asap and I'm going to open a new task if required!